### ISSUE TITLE
static file register arbitrary file read

### ISSUE THREAD

**Issue by yeahx** [channelcat/sanic#633]
File `sanic/sanic/static.py` line 46
we can see here is security check
```
43    async def _handler(request, file_uri=None):
44        # Using this to determine if the URL is trying to break out of the path
45        # served.  os.path.realpath seems to be very slow
46        if file_uri and '../' in file_uri:
47            raise InvalidUsage("Invalid URL")
```
but at the line 56 in this file,the unquote function decoded file_path
`file_path = unquote(file_path)`
we only need to "/" url coding can bypass the previous security check
like this:
![image](https://cloud.githubusercontent.com/assets/11347901/25017430/5ee980ea-20b6-11e7-87b5-8f2999bc33c5.png)

example of vulnerabilities:
```
#!/usr/bin/env python
#-*- coding:utf-8 -*-
#author: lazyago@gmail.com
from sanic import Sanic
from sanic.response import text


app = Sanic()

app.static('/static', '/var/tmp')

@app.route("/")
async def test(request):
    return text('Hello!')

if __name__ == '__main__':
    app.run(host='127.0.0.1', port=8787, debug=True)
```

**Comment by go2sh** [comment]
Autsch.

**Comment by seemethere** [comment]
Closed with #635
